extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQFaultsTest.java

```diff
@@ -356,4 +359,44 @@ public void testTooManyInputFiles() throws IOException
         .setExpectedMSQFault(new TooManyInputFilesFault(numFiles, Limits.MAX_INPUT_FILES_PER_WORKER, 2))
         .verifyResults();
   }
+
+  @Test
+  public void testUnionAllIsDisallowed()
+  {
+    final RowSignature rowSignature =
+        RowSignature.builder().add("__time", ColumnType.LONG).build();
+    testIngestQuery()
+        .setSql("SELECT * FROM foo\n"
+                + "UNION ALL\n"
+                + "SELECT * FROM foo\n")
+        .setExpectedRowSignature(rowSignature)
+        .setExpectedDataSource("foo1")
+        .setExpectedMSQFault(QueryNotSupportedFault.instance())
```

Comment:
Hmm… I am confused about why this yields a `QueryNotSupportedFault`. Shouldn't it fail to plan, and generate a planner error instead of an MSQ fault?

Reply:
It does plan using the `UnionDataSource`, which then goes into MSQ.

Reply:
Added a comment on how it is getting planned.

Reply:
I guess this test can be removed. It should be planned using a UnionDataSource. 
Can we also add the NativeQuery for assertion ?